src/app/zap-templates/zcl/bridged-device-basic.xml

```diff
@@ -0,0 +1,58 @@
+        <attribute side="server" code="0x0007" define="HARDWARE_VERSION" type="INT16U" min="0x0000" max="0xFFFF" writable="false" default="0x00" optional="true">HardwareVersion</attribute>
+        <attribute side="server" code="0x0008" define="HARDWARE_VERSION_STRING" type="CHAR_STRING" length="64" writable="false" default="" optional="true">HardwareVersionString</attribute>
+        <attribute side="server" code="0x0009" define="SOFTWARE_VERSION" type="INT32U" min="0x0000" max="0xFFFF" writable="false" default="0x00" optional="true">SoftwareVersion</attribute>
+        <attribute side="server" code="0x000A" define="SOFTWARE_VERSION_STRING" type="CHAR_STRING" length="64" writable="false" default="" optional="true">SoftwareVersionString</attribute>
+        <attribute side="server" code="0x000B" define="MANUFACTURING_DATE" type="CHAR_STRING" length="16" writable="false" default="" optional="true">ManufacturingDate</attribute>
+        <!-- TODO The actual length is not defined in the spec. But the current implementation expect a length -->
+        <attribute side="server" code="0x000C" define="PART_NUMBER" type="CHAR_STRING" length="254" writable="false" optional="true">PartNumber</attribute>
+        <!-- TODO The spec says the length is 256 ASCII characters. But the current implementation does not support values bigger than 254 -->
+        <attribute side="server" code="0x000D" define="PRODUCT_URL" type="CHAR_STRING" length="254" writable="false" optional="true">ProductURL</attribute>
+        <attribute side="server" code="0x000E" define="PRODUCT_LABEL" type="CHAR_STRING" length="64" writable="false" default="" optional="true">ProductLabel</attribute>
+        <attribute side="server" code="0x000F" define="SERIAL_NUMBER" type="CHAR_STRING" length="32" writable="false" default="" optional="true">SerialNumber</attribute>
+        <attribute side="server" code="0x0011" define="REACHABLE" type="BOOLEAN" min="0x00" max="0x01" writable="false" default="0x00" optional="false">Reachable</attribute>
+        <command source="server" code="0x00" name="StartUp" optional="true">
+        <description>The StartUp event SHALL be emitted by a Node as soon as reasonable after completing a boot or reboot process.</description>
+        </command>
+        <command source="server" code="0x01" name="ShutDown" optional="true">
+        <description>The ShutDown event SHOULD be emitted by a Node prior to any orderly shutdown sequence on a best-effort basis.</description>
+        </command>
+        <command source="server" code="0x02" name="Leave" optional="true">
+        <description>The Leave event SHOULD be emitted by a Node prior to permanently leaving the Fabric.</description>
+        </command>
+        <command source="server" code="0x03" name="ReachableChanged" optional="false">
+        <description>The Leave event SHOULD be emitted by a Node when the Reachable attribute changes.</description>
+        </command>
```

Comment:
These are events, not commands, in the spec...  @vivien-apple are we representing events as commands for now?

Reply:
There doesn't appear to be a way to define events -- yet. I found that many of the new clusters added have omitted the events altogether. However, the BASIC cluster has defined events in this way